nixos/modules/system/boot/systemd.nix

```diff
@@ -201,8 +201,13 @@ let
     ];
 
   makeJobScript = name: text:
-    let mkScriptName =  s: "unit-script-" + (replaceChars [ "\\" "@" ] [ "-" "_" ] (shellEscape s) );
-    in  pkgs.writeTextFile { name = mkScriptName name; executable = true; inherit text; };
+    let
+      scriptName = replaceChars [ "\\" "@" ] [ "-" "_" ] (shellEscape name);
+      out = pkgs.writeShellScriptBin scriptName ''
```

Comment:
`writeShellScriptBin` will include the shebang, so you can remove it from `makeJobScript` invocations below

Reply:
Oh, yes, I meant to remove it, thanks.